imap_processing/tests/spice/test_pointing_frame_handler.py

```diff
@@ -0,0 +1,177 @@
+"""Tests Pointing Frame Generation."""
+
+import shutil
+import sys
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pytest
+import spiceypy as spice
+
+from imap_processing.spice.pointing_frame_handler import (
+    average_quaternions,
+    create_pointing_frame,
+    create_rotation_matrix,
+    get_et_times,
+)
+
+
+@pytest.fixture()
+def kernel_path(tmp_path):
+    """Create path to kernels."""
+
+    test_dir = (
+        Path(sys.modules[__name__.split(".")[0]].__file__).parent
+        / "tests"
+        / "spice"
+        / "test_data"
+    )
+
+    kernels = [
+        "naif0012.tls",
+        "imap_science_0001.tf",
+        "imap_sclk_0000.tsc",
+        "imap_wkcp.tf",
+        "imap_spin.bc",
+    ]
+
+    for file in test_dir.iterdir():
+        if file.name in kernels:
+            shutil.copy(file, tmp_path / file.name)
+
+    return tmp_path
+
+
+@pytest.fixture()
+def create_kernel_list(kernel_path):
+    """Create kernel lists."""
+    kernels = [str(file) for file in kernel_path.iterdir()]
+    ck_kernel = [
+        str(file) for file in kernel_path.iterdir() if file.name == "imap_spin.bc"
+    ]
+
+    return kernels, ck_kernel
+
+
+@pytest.fixture()
+def et_times(create_kernel_list):
+    """Tests get_et_times function."""
+    kernels, ck_kernel = create_kernel_list
+
+    with spice.KernelPool(kernels):
+        et_start, et_end, et_times = get_et_times(str(ck_kernel[0]))
+
+    return et_times
+
+
+@pytest.mark.xfail(reason="Will fail unless kernels in pointing_frame/test_data.")
```

Comment:
This should be a `skipif`, because you want it to actually pass/fail if you have the data there, whereas it will be XPASS in this case but still not count for/against your test coverage.

You can make a decorator at the top of the file that is something like `requires_spice_data = pytest.mark.skipif(not Path(file).exists())` and then mark your tests here with `@requires_spice_data`